src/main/kotlin/calculator/CustomStringDelimiter.kt

```diff
@@ -0,0 +1,50 @@
+package calculator
+
+import kotlin.streams.toList
+
+class CustomStringDelimiter(value: String?) {
+
+    private val values: List<String>
+    private val operands: MutableList<Int>
+    private val operators: MutableList<String>
+
+    init {
+
+        values = validate(value)
+        operators = values.stream()
+            .filter { Expression.STRING_OPERATORS.contains(it) }
+            .toList()
+            .toMutableList()
+
+        operands = values.stream()
+            .filter { !Expression.STRING_OPERATORS.contains(it) }
+            .map { it.toInt() }
+            .toList()
+            .toMutableList()
+    }
+
+    private fun validate(value: String?): List<String> {
+        if (value == null) {
+            throw IllegalArgumentException("잘못된 값이 전달되었습니다.")
+        }
+
+        if (value.isBlank() || value.isEmpty()) {
+            throw IllegalArgumentException("잘못된 값이 전달되었습니다.")
+        }
```

Comment:
```suggestion
       if (value.isNullOrBlank()) {
            throw IllegalArgumentException("잘못된 값이 전달되었습니다.")
        }
```

`isNullOrBlank` 로 처리한다면 `null, blank, empty` 모두 체크를 할 수가 있습니다 ☺️

참고
 - [isNullOrBlank / isNullOrEmpty](https://codechacha.com/ko/kotlin-string-null-empty-check/)


------

추가로 kotlin에서는 `require` 함수를 통해서 `IllegalArgumentException`을 처리할 수 있습니다!

ex ) 
`require(value.isNullOrBlank()) {"잘못된 값이 전달되었습니다."}` 

참고
- [Kotlin 예외처리](https://seosh817.tistory.com/155)